src/deadline/job_attachments/upload.py

```diff
@@ -382,6 +392,78 @@ def upload_input_files(
                     "File upload cancelled.", progress_tracker.get_summary_statistics()
                 )
 
+    def reset_s3_check_cache(self, s3_check_cache_dir: Optional[str]) -> None:
+        """
+        Resets the S3 check cache by removing the cache altogether.
+        """
+        with S3CheckCache(s3_check_cache_dir) as s3_check_cache:
+            logger.debug(
+                f"The s3_check_cache.db file in {s3_check_cache_dir} will be deleted, "
+                f"as a mismatch between the cache and the actual hash in S3 was found"
+            )
+            # Remove the cache file
+            s3_check_cache.remove_cache()
+
+    def _check_hashes_exist_in_s3(self, cache_entries: List[S3CheckCacheEntry]) -> bool:
+        """
+        checks if the hashes in the cache entries exist in S3
+        """
+        for cache_entry in cache_entries:
+            try:
+                # Split the S3 key into bucket and key parts
+                bucket, key = cache_entry.s3_key.split("/", 1)
+
+                # Check if the object is already uploaded and exist in S3 bucket
+                if self.file_already_uploaded(bucket=bucket, key=key):
+                    logger.debug(f"cache_entry: {cache_entry} exist in S3")
+
+                # If a mismatch found, return False to reset the cache immediately. There's no need to check the rest.
+                else:
+                    return False
+            except Exception as e:
+                # If error occurs, log warning and return False to indicate cache reset required
+                logger.warning(f"Error occurred while checking {cache_entry}. Exception: {e}")
+                return False
+
+        # Otherwise all hashes exist in S3
+        return True
+
+    def _sample_cache_entries_with_limit(
+        self, cache_entries: List[S3CheckCacheEntry], limit: int = 30
+    ):
+        sampled_count: int = min(len(cache_entries), limit)
+        random.shuffle(cache_entries)
+        return cache_entries[:sampled_count]
```

Comment:
nit: does this need to be a separate method? maybe we can refactor it into a method when we need to reuse it